library/x509.c

```diff
@@ -138,31 +140,31 @@ static inline const char* md_type_to_string( mbedtls_md_type_t md_alg )
 {
     switch( md_alg )
     {
-#if defined(MBEDTLS_MD5_C)
```

Comment:
Minor: This data would in my opinion fall into the `VIA_LOWLEVEL_OR_PSA` category, but x509 will be defined only with `MBEDTLS_USE_PSA_CRYPTO` or `MBEDTLS_MD_C`, so that's not a problem.

Reply:
I think MD type refers to MD layer.

Reply:
I know, what I mean is that the strings below are just metadata, and could be useful without the `MBEDTLS_MD_C` module, so instead of using, for example `MBEDTLS_HAS_ALG_SHA_1_VIA_MD_OR_PSA`, `MBEDTLS_HAS_ALG_SHA_1_VIA_LOWLEVEL_OR_PSA` could be used here, but the x509 module itself requires either `MBEDTLS_USE_PSA_CRYPTO` or `MBEDTLS_MD_C`, so it doesn't make a difference - this code will not be used without `MBEDTLS_MD_C` and without `MBEDTLS_PSA_CRYPTO_C`.